### ISSUE TITLE
Incorrect typing (and doc) for cy.server() command

### ISSUE THREAD

**Issue by bahmutov** [cypress-io/cypress#1907]
Documentation says that `cy.server()` yields `null`, but it yields default options

https://docs.cypress.io/api/commands/server.html#Yields

```js
cy.server().should((server) => {
      // the default options on server
      // you can override any of these options
      expect(server.delay).to.eq(0)
      expect(server.method).to.eq('GET')
      expect(server.status).to.eq(200)
      expect(server.headers).to.be.null
      expect(server.response).to.be.null
      expect(server.onRequest).to.be.undefined
      expect(server.onResponse).to.be.undefined
      expect(server.onAbort).to.be.undefined

      // These options control the server behavior
      // affecting all requests

      // pass false to disable existing route stubs
      expect(server.enable).to.be.true
      // forces requests that don't match your routes to 404
      expect(server.force404).to.be.false
      // whitelists requests from ever being logged or stubbed
      expect(server.whitelist).to.be.a('function')
    })
```

**Comment by jennifer-shehane** [comment]
Talking with @brian-mann about this, he mentioned that the implementation of `cy.server()` should yield `null`, so we should enforce that implementation as opposed to documenting it and updating the typescript typings.

**Comment by brian-mann** [comment]
It appears this was already merged in unbeknownst to me, since the PR did not close this issue. So we'll roll with these typescript changes and yank them out later when we refactor the network stubbing layer.

Fixed by https://github.com/cypress-io/cypress/pull/1911

**Comment by brian-mann** [comment]
Released in [`3.0.2`](https://docs.cypress.io/guides/references/changelog.html#3-0-2).
